feat(ui): add burger menu toggle for collapsible sidebar with responsive and persistent behavior

yaxit24 opened the pull request:
Fixes: https://github.com/fossasia/eventyay-tickets/issues/727

https://github.com/user-attachments/assets/33d855d9-6eba-4762-b975-dfd88c66c221

## Summary by Sourcery

Implement a burger menu toggle to collapse and expand the left sidebar with responsive behavior and persistent state across desktop and mobile.

Enhancements:
- Add a sidebar toggle button (burger menu) in base templates for control, common, and admin pages
- Implement JavaScript logic to collapse/expand the sidebar with state persistence in localStorage and responsive off-canvas behavior on mobile
- Enhance submenu interactions with accordion-style mobile behavior, click-through support, and hover-to-expand on desktop
- Introduce SCSS styles for the minimized sidebar state, mobile transitions, and minor navbar adjustments
- Bump jQuery to 3.7.1 and include Stripe v3 script in the common base template

Gagan-Ram reviewed (changes requested):
https://github.com/user-attachments/assets/d8588089-4f5f-4d84-8651-23eb709fd4ea

Hi @yaxit24, I noticed the following issues:
1. The burger menu gets toggled off when navigating to a new page.
2. The alignment of the burger menu items doesn't match that of ey-talk. Specifically, the top menu item icon appears to have extra padding and is aligned further left compared to the icons of the items below it.
3. The entire Tickets component is displaying an "Internal Server Error".
4. The burger menu icon is not visible on the Tickets component pages.

Also, please clarify why have you changed "Main dashboard" to "Tickets dashboard" and modified its links?

hongquan commented on src/pretix/eventyay_common/templates/eventyay_common/base.html:
```diff
@@ -114,37 +118,90 @@
                             </form>
                         {% else %}
-                            <a href="{% eventurl request.event 'presale:event.index' %}" title="{% translate "View event" %}"
+                            <a href="{% eventurl request.event "presale:event.index" %}" title="{% trans "View event" %}"
```
Why you change the _correct_ quote styles to _wrong_ ones?

hongquan reviewed (changes requested):
You broke all quote styles.
The rule is to use different quote styles when nesting. Now you apply the same quote style everywhere!

yaxit24 commented:
please verify: 

Uploading Screen Recording 2025-07-08 at 10.10.01 AM.mov…

Uploading Screen Recording 2025-07-08 at 10.13.45 AM.mov…

mariobehling reviewed (changes requested):
I dont understand the logic of the mobile view. Why should we have two burger menus in the mobile view? Please implement it in the same way as eventyay-talk. One mobile menu on the left is enough.

yaxit24 commented:
<img width="655" height="657" alt="Screenshot 2025-07-10 at 4 21 58 PM" src="https://github.com/user-attachments/assets/28ce867a-3580-4bde-9b97-cc95cee920b8" />
Thanks, implemented. I had added the functionality to both burger buttons to enhance accessibility on mobile devices, allowing them to open and close the sidebars from either side of the mobile view.

Gagan-Ram reviewed (changes requested):
1. The alignment of the burger menu items doesn't match that of ey-talk. Specifically, the top menu item icon appears to have extra padding and is aligned further left compared to the icons of the items below it.
2. If a sidebar menu item contains a sub-menu, selecting any sub-menu item will cause the entire dropdown menu to collapse.

mariobehling commented:
As there were no updates on this PR I am closing it.

mariobehling reviewed (changes requested):
> * The alignment of the burger menu items doesn't match that of ey-talk. Specifically, the top menu item icon appears to have extra padding and is aligned further left compared to the icons of the items below it.
> * If a sidebar menu item contains a sub-menu, selecting any sub-menu item will cause the entire dropdown menu to collapse.

In the video it still looks like these points have not been resolved.

mariobehling reviewed (changes requested):
I removed assignees as "assignees" means that they have to take care of the PR. Instead we already have reviewers. 

Please follow up and resolve the outstanding change requests when possible. Thanks.

yaxit24 commented:
https://github.com/user-attachments/assets/741fd1cc-424d-425b-83eb-974c1cae2afa

yaxit24 commented on app/eventyay/static/pretixcontrol/js/sb-admin-2.js:
```diff
@@ -10,15 +10,84 @@
+    function isDesktopView() {
+        return window.matchMedia("(min-width: 1025px)").matches;
+    }
+
```
Applied

yaxit24 commented on app/eventyay/static/pretixcontrol/js/sb-admin-2.js:
```diff
@@ -29,7 +98,12 @@ $(function () {
+    function stopPropagationHandler(e) {
+        e.stopPropagation();
+    }
+    [$sidebar, $pageWrapper].forEach(function($el) {
```
Applied

Gagan-Ram reviewed (approved):
Works as expected.
Please address copilot suggestions.

yaxit24 commented on app/eventyay/static/pretixcontrol/scss/_sidebar.scss:
```diff
@@ -0,0 +1,384 @@
+    }
+  }
+  
+  body.sidebar-minimized:has(.sidebar:hover) #page-wrapper {
```
I have tested it on Safari , Chrome , Firefox everything works perfectly.

yaxit24 commented on app/eventyay/static/pretixcontrol/js/sb-admin-2.js:
```diff
@@ -10,15 +10,78 @@
+    function updateCSSVariables() {
+        document.documentElement.style.setProperty('--navbar-height', getNavbarHeight() + 'px');
+    }
+    const $sidebarToggleButton = $('#sidebar-toggle');
```
Not necessary, moving it to the top with the other jQuery selections improves consistency and readability, but does not change behavior.
